git-backdate

```diff
@@ -82,32 +91,34 @@ def rewrite_history(
     elif no_business_hours:
         min_hour = 18
         max_hour = 23
-    duration = len(days)
-    commits_per_day = math.ceil(len(commits) / duration)
-    last_timestamp = None
-    commit_count = len(commits)
-    day_progress = 0
-    for commit_index in range(commit_count):
-        progress = (commit_index + 1) / commit_count
+    prev_timestamp = prev_commit_timestamp(commits)
+    commit_total_count = len(commits)
+    daily_commit_limit = math.ceil(commit_total_count / len(days))
+    daily_commit_count = 0
+    for commit_index in range(commit_total_count):
+        commit_progress = (commit_index + 1) / commit_total_count
         # first, choose the date
-        date_index = round(progress * (duration - 1))
-        date = days[date_index]
-        if not last_timestamp or date == last_timestamp.date():
-            day_progress += 1
+        day_index = round(commit_progress * (len(days) - 1))
+        date = days[day_index]
+
+        if not prev_timestamp or date == prev_timestamp.date():
+            daily_commit_count += 1
+            _min_hour = min_hour if min_hour > prev_timestamp.hour else prev_timestamp.hour
         else:
-            day_progress = 0
+            daily_commit_count = 0
+            _min_hour = min_hour
```

Comment:
Really sorry to be grumpy, but this place illustrates beautifully how short and readable this PR could have been without the renaming of variables.